libraries/default.rb

```diff
@@ -0,0 +1,62 @@
+#
+# Cookbook Name:: kernel-modules
+# Author:: Jeremy MAURO <j.mauro@criteo.com>
+#
+# Copyright 2016, Criteo.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+module KernelModules
+  # Function for handling the modprobe format
+  def self.modprobe_config(name, config)
+    # Idempotency
+    output = []
+    config.keys.sort.each do |cmd|
+      value = config[cmd]
+      output << case value
+                # Ex:
+                #  alias: "nfs4 nfs"
+                when String
+                  if cmd.to_s == 'alias'
+                    [cmd, value, name].join(' ')
+                  else
+                    # "value" could be an empty string ''
+                    value.empty? ? [cmd, name].join(' ') : [cmd, name, value].join(' ')
+                  end
+                # Ex:
+                #  options: [
+                #    'enable_ino64=1',
+                #    'nfs4_disable_idmapping=0',
+                #  ]
+                when Array
+                  if cmd.to_s == 'alias'
+                    value.sort.map { |v| "#{cmd} #{v} #{name}" }.join("\n")
+                  else
+                    value.sort.map { |v| "#{cmd} #{name} #{v}" }.join("\n")
+                  end
+                else
+                  [cmd, name].join(' ')
```

Comment:
Which case is this? boolean?

I think that this whole method should have more comments with examples of module configs.